applications/FluidDynamicsApplication/python_scripts/apply_mass_conservation_check_process.py

```diff
@@ -46,20 +75,98 @@ def ExecuteInitialize(self):
+
+        ### writing first line in file
+        if (self._fluid_model_part.ProcessInfo[KratosMultiphysics.STEP] % self._correct_frequency == 0):
+            if ( self._write_to_log and self._is_printing_rank ):
+                with open(self._my_log_file, "a+") as logFile:
+                    logFile.write( log_line_string )
+
+            # (1) #
+            if ( self._perform_local_corr ):
+                ### perform the local conservation procedure
+                # REMARK: The pseudo time step dt has no meaning as a physical time step and is NOT accounted as such.
+                # Instead, it as purely used for an artificial convection that helps to conserve the mass.
+                # Accordingly, dt depends on the currently necessary mass convection.
+                dt = self.mass_conservation_utility.ComputeDtForConvection()
+
+                # REMARK: The distance field is artificially convected to extrapolate the interface motion into the future.
+                # The time step considered for this artificial "forward convection" is dt.
+                # The previous distance field is saved in an auxiliary variable
+                prev_dt = self._fluid_model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME]
+                self._fluid_model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME] = dt
+                KratosMultiphysics.VariableUtils().SaveScalarVar(
+                    KratosMultiphysics.DISTANCE,
+                    KratosFluid.AUX_DISTANCE,
+                    self._fluid_model_part.GetCommunicator().LocalMesh().Nodes)
+                self.forward_convection_process.Execute()
+                self._fluid_model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME] = prev_dt
+
+                # REMARK: A comparison between the "forward_convected" distance field and the current distance field is made.
+                # Depending on the current mass balance, one of the following options i chosen:
+                # - taking min() of both fields = adding to fluid_1 ("more water")
+                # - taking max() of both fields = adding to fluid_2 ("more air")
+                self.mass_conservation_utility.ApplyLocalCorrection( KratosFluid.AUX_DISTANCE )
+
+            # (2) #
+            if ( self._perform_local_corr and self._perform_global_corr ):
+                ### check how much work has already been done by the local correction process
+                self.mass_conservation_utility.ReCheckTheMassConservation()
+
+            # (3) #
+            if ( self._perform_global_corr ):
+                ### perform the global correction
+                # without any consideration of the velocity field, volume is added by a global slight shift of the distance field
+                self.mass_conservation_utility.ApplyGlobalCorrection()
+
+
+    def _set_levelset_convection_process(self):
+        if KratosMultiphysics.ParallelEnvironment.GetDefaultDataCommunicator().IsDistributed():
```

Comment:
you should check your ModelPart if it is distributed and not if Kratos runs in MPI

Reply:
ok